File: sdks/local-rp/csharp/src/LinkKeys.LocalRp/Crypto/CryptoException.cs
namespace LinkKeys.LocalRp.Crypto;

/// <summary>
/// Thrown for any cryptographic failure in this namespace: signature verification
/// failure, AEAD authentication failure, a non-contributory (low-order) X25519 key,
/// a malformed key length, or an unexpected NSec/BCL provider error. Never carries
/// key material, shared secrets, or plaintext in its message (AGENTS.md: "Never log
/// sensitive information").
/// </summary>
public class CryptoException : Exception
{
    public CryptoException(string message) : base(message)
    {
    }

    public CryptoException(string message, Exception cause) : base(message, cause)
    {
    }
}

File: sdks/local-rp/java/src/main/java/community/catalyst/linkkeys/localrp/crypto/CryptoException.java
package community.catalyst.linkkeys.localrp.crypto;

/**
 * Thrown for any cryptographic failure in this package: signature
 * verification failure, AEAD authentication failure, a non-contributory
 * (low-order) X25519 key, a malformed key length, or an unexpected JCA
 * provider error. Never carries key material, shared secrets, or plaintext
 * in its message (AGENTS.md: "Never log sensitive information").
 */
public class CryptoException extends RuntimeException {
    public CryptoException(String message) {
        super(message);
    }

    public CryptoException(String message, Throwable cause) {
        super(message, cause);
    }
}
